clang/Diagnostic.d

```diff
@@ -125,3 +125,30 @@ struct DiagnosticSet
         return length;
     }
 }
+
+CXDiagnosticSeverity severity(DiagnosticSet diagnostics)
+{
+    import std.algorithm.searching : minPos;
+    import std.algorithm.iteration : map;
+
+    bool less(CXDiagnosticSeverity a, CXDiagnosticSeverity b)
+    {
+        return cast(uint) a > cast(uint) b;
+    }
```

Comment:
I think using a nested function will cause an unnecessary closure. You can use static function, or even better an alias instead https://dlang.org/changelog/2.070.0.html#alias-funclit.